source/using-keycloak.rst

```diff
@@ -0,0 +1,392 @@
+   one lowercase letter and one special character.
+
+
+Docker compose
+^^^^^^^^^^^^^^
+
+You need to set a password value in the *.env* file for the
+``KEYCLOAK_MASTER_PASSWORD`` and ``KEYCLOAK_ADMIN_PASSWORD_TEMP``
+variables. The ``KEYCLOAK_ADMIN_PASSWORD_TEMP`` value will be the *admin*
+user password of all realms when created.
+See more at `settings required`_.
+
+
+Ansible (kubernetes)
+^^^^^^^^^^^^^^^^^^^^
+
+*Work in progress*
+
+Setting SMTP via environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to configure SMTP via environment variables,
+but beware these values are only applied when the *realm*
+is created, it will not affect the existing realms.
+To configure a *realm* that already exists,
+see `Configuring SMTP through the graphical interface`.
+
+
+Docker compose
+^^^^^^^^^^^^^^
+
+See more at `configuring SMTP`_.
+
+Ansible (kubernetes)
+^^^^^^^^^^^^^^^^^^^^
+
+*Work in progress*
+
+Setting HTTPs as required via environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note: These values are only applied when the *realm* is created,
+it will not affect existing realms.
+For existing reals see `Configuring HTTPs as required`
+
+Docker compose
+^^^^^^^^^^^^^^
+
+You need to set the **EXTERNAL** value in the *.env*
+file to the ``KEYCLOAK_REALM_SSL_MODE`` variable. See more at `configuring HTTPs`_.
+
+Ansible (kubernetes)
+^^^^^^^^^^^^^^^^^^^^
+
+*Work in progress*
+
+.. _Base keycloak configuration file:
+
+Base keycloak configuration file for creating realms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Configuration file used as the basis for creating realms in keycloak.
+It is the file exported from a realm by the full keycloak.
+This file is configured using the ``DOJOT_CUSTOM_REALM_REP_FILE``
+environment variable in the keycloak service.
+When creating a new realm the identifiers (*ids*) and
+realm name in this file are ignored. However, it is important
+to point out that our *deployments* for both *docker-compose*
+and *ansible (kubernetes)* already have a file with the necessary settings to use dojot.
+
+Docker compose
+^^^^^^^^^^^^^^
+
+In the case of docker-compose this file is in `keycloak/customRealmRepresentation.json`
+
+Ansible (kubernetes)
+^^^^^^^^^^^^^^^^^^^^
+
+*Work in progress*
+
+Old libraries
+~~~~~~~~~~~~~
+
+To maintain compatibility with dojot it is necessary to provide the login
+and password of the master or a user who can get the list of realms
+available in the older libraries, namely `dojot-module-nodejs`_,
+`iotagent-nodejs`_, `dojot-module-python`_,  `dojot-module-java`_,
+`iotagent-java`_, see more in their respective documentation.
+However, it is important to point out that our *deployments* for
+both *docker-compose* and *ansible (kubernetes)* are already prepared
+for this and it is only necessary to configure the passwords as described
+in the topic `Setting Passwords (Required) via environment variables`.
```

Comment:
I don't know if it should be a link in `Setting Passwords (Required) via environment variables`, but if it should, the link was not set.